bluetooth_proxy on  esp-idf framework compile error

### The problem

Trying to setup bluetooth proxy on Wemos S2 mini (ESP32S2)

After including
```
esp32_ble_tracker:
bluetooth_proxy:
```
to config file  got compile error

```
src/esphome/components/esp32_ble_tracker/queue.h:11:10: fatal error: esp_gap_ble_api.h: No such file or directory
```


### Which version of ESPHome has the issue?

2022.8.3

### What type of installation are you using?

Home Assistant Add-on

### Which version of Home Assistant has the issue?

2022.09

### What platform are you using?

ESP32-IDF

### Board

Wemos S2 mini 

### Component causing the issue

bluetooth_proxy

### Example YAML snippet

```yaml
esphome:
  name: proxy

esp32:
  board: esp32-s2-saola-1
  framework:
    type: esp-idf


logger:
api:
ota:
wifi:
  ssid: "SSID"
  password: "PASSWORD"
  
esp32_ble_tracker:
bluetooth_proxy:
```


### Anything in the logs that might be useful for us?

```txt
INFO Reading configuration /config/esphome/proxy.yaml...
INFO Generating C++ source...
INFO Compiling app...
Processing proxy (board: esp32-s2-saola-1; framework: espidf; platform: platformio/espressif32 @ 3.5.0)
--------------------------------------------------------------------------------
HARDWARE: ESP32S2 240MHz, 320KB RAM, 4MB Flash
 - framework-espidf @ 3.40302.0 (4.3.2) 
 - tool-cmake @ 3.16.4 
 - tool-ninja @ 1.7.1 
 - toolchain-riscv32-esp @ 8.4.0+2021r2-patch2 
 - toolchain-xtensa-esp32s2 @ 8.4.0+2021r2-patch2
Reading CMake configuration...
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
No dependencies
Compiling /data/proxy/.pioenvs/proxy/src/esphome/components/api/api_connection.o
Compiling /data/proxy/.pioenvs/proxy/src/esphome/components/api/api_server.o
Compiling /data/proxy/.pioenvs/proxy/src/esphome/components/api/list_entities.o
Compiling /data/proxy/.pioenvs/proxy/src/esphome/components/api/proto.o
In file included from src/esphome/components/esp32_ble_tracker/esp32_ble_tracker.h:5,
                 from src/esphome/components/bluetooth_proxy/bluetooth_proxy.h:7,
                 from src/esphome/components/api/api_connection.h:11,
                 from src/esphome/components/api/api_connection.cpp:1:
src/esphome/components/esp32_ble_tracker/queue.h:11:10: fatal error: esp_gap_ble_api.h: No such file or directory

*************************************************************************
* Looking for esp_gap_ble_api.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:esp_gap_ble_api.h"
* Web  > https://registry.platformio.org/search?q=header:esp_gap_ble_api.h
*
*************************************************************************

 #include <esp_gap_ble_api.h>
          ^~~~~~~~~~~~~~~~~~~
compilation terminated.
*** [/data/proxy/.pioenvs/proxy/src/esphome/components/api/api_connection.o] Error 1
In file included from src/esphome/components/esp32_ble_tracker/esp32_ble_tracker.h:5,
                 from src/esphome/components/bluetooth_proxy/bluetooth_proxy.h:7,
                 from src/esphome/components/api/api_connection.h:11,
                 from src/esphome/components/api/api_server.cpp:2:
src/esphome/components/esp32_ble_tracker/queue.h:11:10: fatal error: esp_gap_ble_api.h: No such file or directory

*************************************************************************
* Looking for esp_gap_ble_api.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:esp_gap_ble_api.h"
* Web  > https://registry.platformio.org/search?q=header:esp_gap_ble_api.h
*
*************************************************************************

 #include <esp_gap_ble_api.h>
          ^~~~~~~~~~~~~~~~~~~
compilation terminated.
*** [/data/proxy/.pioenvs/proxy/src/esphome/components/api/api_server.o] Error 1
In file included from src/esphome/components/esp32_ble_tracker/esp32_ble_tracker.h:5,
                 from src/esphome/components/bluetooth_proxy/bluetooth_proxy.h:7,
                 from src/esphome/components/api/api_connection.h:11,
                 from src/esphome/components/api/list_entities.cpp:5:
src/esphome/components/esp32_ble_tracker/queue.h:11:10: fatal error: esp_gap_ble_api.h: No such file or directory

*************************************************************************
* Looking for esp_gap_ble_api.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:esp_gap_ble_api.h"
* Web  > https://registry.platformio.org/search?q=header:esp_gap_ble_api.h
*
*************************************************************************

 #include <esp_gap_ble_api.h>
          ^~~~~~~~~~~~~~~~~~~
compilation terminated.
*** [/data/proxy/.pioenvs/proxy/src/esphome/components/api/list_entities.o] Error 1
========================== [FAILED] Took 4.02 seconds ==========================
```


### Additional information

_No response_

--- a/esphome/components/esp32_ble/__init__.py
+++ b/esphome/components/esp32_ble/__init__.py
@@ -2,11 +2,13 @@ import esphome.codegen as cg
 import esphome.config_validation as cv
 from esphome.const import CONF_ID
 from esphome.core import CORE
-from esphome.components.esp32 import add_idf_sdkconfig_option
+from esphome.components.esp32 import add_idf_sdkconfig_option, get_esp32_variant, const
 
 DEPENDENCIES = ["esp32"]
 CODEOWNERS = ["@jesserockz"]
 CONFLICTS_WITH = ["esp32_ble_tracker", "esp32_ble_beacon"]
+
+NO_BLUTOOTH_VARIANTS = [const.VARIANT_ESP32S2]
 
 esp32_ble_ns = cg.esphome_ns.namespace("esp32_ble")
 ESP32BLE = esp32_ble_ns.class_("ESP32BLE", cg.Component)
@@ -17,6 +19,15 @@ CONFIG_SCHEMA = cv.Schema(
         cv.GenerateID(): cv.declare_id(ESP32BLE),
     }
 ).extend(cv.COMPONENT_SCHEMA)
+
+
+def validate_variant(_):
+    variant = get_esp32_variant()
+    if variant in NO_BLUTOOTH_VARIANTS:
+        raise cv.Invalid(f"{variant} does not support Bluetooth")
+
+
+FINAL_VALIDATE_SCHEMA = validate_variant
 
 
 async def to_code(config):

--- a/esphome/components/esp32_ble_beacon/__init__.py
+++ b/esphome/components/esp32_ble_beacon/__init__.py
@@ -3,6 +3,7 @@ import esphome.config_validation as cv
 from esphome.const import CONF_ID, CONF_TYPE, CONF_UUID, CONF_TX_POWER
 from esphome.core import CORE, TimePeriod
 from esphome.components.esp32 import add_idf_sdkconfig_option
+from esphome.components import esp32_ble
 
 DEPENDENCIES = ["esp32"]
 CONFLICTS_WITH = ["esp32_ble_tracker"]
@@ -53,6 +54,8 @@ CONFIG_SCHEMA = cv.All(
     ).extend(cv.COMPONENT_SCHEMA),
     validate_config,
 )
+
+FINAL_VALIDATE_SCHEMA = esp32_ble.validate_variant
 
 
 async def to_code(config):

--- a/esphome/components/esp32_ble_tracker/__init__.py
+++ b/esphome/components/esp32_ble_tracker/__init__.py
@@ -17,6 +17,7 @@ from esphome.const import (
 )
 from esphome.core import CORE
 from esphome.components.esp32 import add_idf_sdkconfig_option
+from esphome.components import esp32_ble
 
 DEPENDENCIES = ["esp32"]
 
@@ -186,6 +187,8 @@ CONFIG_SCHEMA = cv.Schema(
         ),
     }
 ).extend(cv.COMPONENT_SCHEMA)
+
+FINAL_VALIDATE_SCHEMA = esp32_ble.validate_variant
 
 ESP_BLE_DEVICE_SCHEMA = cv.Schema(
     {

